Save renovation distribution

ricardarosemann opened the pull request:
Save the distribution of heating system renovation in input.gdx with high time resolution.

This is (only) needed for reporting purposes in the lifetime and LCC/LCOH analysis.

In addition, I added more names of time dimensions to be converted to numeric in ```expandSets``` and ```readSymbol```. I don't think it makes a difference for ```readSymbol```, but for ```expandSets``` I think it can be useful.

In principle, it would also fit nicely to include the renaming of ttot2 -> ttotIn and ttot -> ttotOut in this PR, because in the R-code, this appears precisely in the context of the lifetime computations. I haven't done it yet, though, because it was quite some effort in ```reportbrick```, so I'm a bit undecided whether I want to invest the time.

robinhasse commented on R/createParameters.R:
```diff
@@ -387,6 +400,41 @@ createParameters <- function(m, config, inputDir) {
+
+  timePeriodsFullInit <- data.frame(
+    ttot2 = as.numeric(as.character(readSymbol(m, symbol = "tinit"))),
+    ttot = seq(0, 60, 0.5)
+  ) %>%
+    mutate(ttot2 = .data$ttot2 - standingLifeTimeHs,
+           ttot = .data$ttot2 + .data$ttot)
```
```suggestion
    lt = seq(0, 60, 0.5)
  ) %>%
    mutate(ttot2 = .data$ttot2 - standingLifeTimeHs,
           ttot = .data$ttot2 + .data$lt,
           .keep = "none")
```

robinhasse commented on R/createParameters.R:
```diff
@@ -387,6 +400,41 @@ createParameters <- function(m, config, inputDir) {
+  )
+
+  timePeriodsFullInit <- data.frame(
+    ttot2 = as.numeric(as.character(readSymbol(m, symbol = "tinit"))),
```
I think you don't even need the extra argument as temporal dimensions are transformed to numeric by `readSymbol`.

ricardarosemann replied:
I haven't tried the ```stringAsFactor=FALSE``` variant, but without it, it definitely doesn't work: ```readSymbol``` only transforms columns that have a time dimension as column name. For a one-dimensional set (as ```tinit```) this is not the case. I thought about implementing an additional code bit that transforms one-dimension sets with a time dimension as symbol, but decided against it because we barely would use such a construct. For consistency it might still make sense to implement this - then we could be certain that ```readSymbol``` always returns time dimensions as numeric. What do you think?

ricardarosemann commented on R/createParameters.R:
```diff
@@ -387,6 +400,41 @@ createParameters <- function(m, config, inputDir) {
 
+  # Only for reporting: Weibull distribution with high time resolution
+  timePeriodsFull <- expandSets(ttot2 = "ttot", .m = m) %>%
+    tidyr::crossing(ttot = seq(0, 60, 0.5)) %>%
+    mutate(ttot = .data$ttot2 + .data$ttot)
```
I agree and would think that the improved readability justifies the extra ```select```.

ricardarosemann commented:
I now introduced the following additional changes:
- Rename ttot2 and in some cases also ttot to ```ttotIn``` and ```ttotOut``` where appropriate, both in GAMS and R code
- Modify ```readSymbol``` so that one-dimensional time sets are also converted to numeric.
- Introduce option to ```toModelResolution``` that allows to pass certain dimensions that should be ignored when checking for compliance with the model resolution.
- Minor: Add ```.vscode``` directory to ignore-files.

robinhasse commented on R/createParameters.R:
```diff
@@ -243,52 +243,57 @@ createParameters <- function(m, config, inputDir) {
     share %>%
-      mutate(p  = pweibull(.data$lt, .data$shape, .data$scale),
-             value = (.data$p - .data$p0) / (1 - .data$p0),
+      mutate(p  = pweibull(.data$lt, .data$shape, .data$scale)) %>%
+      mutate(value = (.data$p - .data$p0) / (1 - .data$p0),
```
Is there a reason to start a new mutate call? Did the old code not work?
```suggestion
      mutate(p  = pweibull(.data$lt, .data$shape, .data$scale),
             value = (.data$p - .data$p0) / (1 - .data$p0),
```

robinhasse commented on R/createParameters.R:
```diff
@@ -243,52 +243,57 @@ createParameters <- function(m, config, inputDir) {
   # parameter for monitoring purposes
-  p_probDem <- expandSets("region", "typ", ttot2 = "ttot", "ttot", .m = m) %>%
-    filter(.data[["ttot2"]] >= .data[["ttot"]]) %>%
+  p_probDem <- expandSets("region", "typ", ttotIn = "ttot", ttotOut = "ttot", .m = m) %>%
+    filter(.data[["ttotOut"]] >= .data[["ttotIn"]]) %>%
```
Very minor: If you want to, you could change `.data[["..."]]` to `.data$...` where you change code anyways...

robinhasse commented on R/createParameters.R:
```diff
@@ -366,27 +371,66 @@ createParameters <- function(m, config, inputDir) {
+
+  p_shareRenHSfullInit <- shareRen(
+    ltHs,
+    standingLifeTime = 0, # We already subtracted the standing lifetime, but still want the standing stock case
```
I personally didn't quite understand this despite the comment. So for me it wasn't very helpful. With `standingLifeTime = 0`, you simply remove the 1/2 time step length. I guess that's because you compare stocks with stocks? In any case it would rather help me if you explain _why_ you need the standing stock case.

ricardarosemann replied:
I added some more comments, if you like, you can give some feedback whether this helps you.

robinhasse commented on R/toModelResolution.R:
```diff
@@ -2,28 +2,29 @@
 #' @importFrom quitte interpolate_missing_periods_
 
-toModelResolution <- function(x, m, value = "value") {
+toModelResolution <- function(x, m, value = "value", ignoreDims = NULL) {
```
Im wondering if `ignoreDims` is the best argument name, since the all entries are allowed. Maybe something like `unfilteredDims`? But I'm not sure and can certainly live with this name, too.

ricardarosemann commented:
> One extra thought: For some reason, we add the alias `ttot2` also to the input.gdx in `createSets` (line 49). I'd either add the new aliases, too, or, maybe even better, remove this line entirely as it is not needed.

The reason was that this was necessary for the old implementation of ```toModelResolution```, applied to ```p_shareRen...```. Now it's not anymore, so I removed it.
